### ISSUE TITLE
Connection lost. Attempting to re-establish...

### ISSUE THREAD

**Issue by dkropmanFD** [jamesbarnett91/tplink-energy-monitor#10]
Close this if you think it may be only on my system but I get the following error when using Windows 10 (latest updates applied) and chrome 64 bit version 70.0.3538.67 

Connecting to remote machine xx.xx.xx.xx:3000 I get a constant flashing:

Connection lost. Attempting to re-establish...

This woks on the same machine running Edge browser (I have same issue above with IE)

This also works if I run on localhost with firefox.

I am running the pre-compiled linux download. The app looks like it is working but not from those browsers.

Thanks

**Comment by ballachango** [comment]
This may help... it may not.

Things have always worked for me using IPv4:3000, but also wanted to try hostname:3000 . When I tried this, it would load the page, but never any data, showing instead "Connection lost. Attempting to re-establish..." as above.

However, 5 minutes later, I can't get any response at all using hostname:3000 (hostname:80 correctly loads a regular webpage)

**Comment by jamesbarnett91** [comment]
@dkropmanFD so you are running the app on a linux host and trying to access it from a Windows machine on the same network? And it works in Edge but not Chrome or IE?

I'm doing something similar, running the app on a linux host and accessing it from various deceives (Windows, Linux, iOS) and haven't had any issues accessing by hostname or IP on port 3000.

The websocket connection is made to `'ws://' + window.location.host + '/ws'` so as long as `window.location.host` is returning the correct hostname/port+ip string it should be fine. 

Do you have any other components between the host and the app (i.e. a reverse proxy)? If so sometimes they need to be configured to allow websockets through.

**Comment by dkropmanFD** [comment]
Yes, Edge works but not on IE or Chrome. I've only tested from my laptop. Works on chrome on ipad.

I have no other components between host and app. I will test tomorrow with my desktop accessing the server. BTW, this all worked before the last update (as far as I remember). Here is a screenshot

![image](https://user-images.githubusercontent.com/36878015/47827931-edb83b00-dd56-11e8-96a2-3c44ee059af4.png)

**Comment by jamesbarnett91** [comment]
Sorry for taking ages to reply!
Do you get any JavaScript errors in the browser console? (Pressing F12 or ctrl+shift+i should open the dev tools and let you view the console)
